lib/helpers/signed-aws-es-fetch.js

```diff
@@ -97,28 +93,39 @@ function setAwsCredentials(creds) {
 	return creds;
 }
 
-function getSignableData(url, opts, creds) {
-	creds = setAwsCredentials(creds);
-	const { host, pathname, search, protocol } = getURLObject(url);
-	const path = `${pathname}${search}`;
-	const signable = {
-		method: opts.method,
-		host,
-		path,
-		body: opts.body,
-		headers: opts.headers
-	};
-	aws4.sign(signable, creds);
-	return { headers: signable.headers, path: signable.path, protocol };
+async function getSignableData(url) {
+	const urlObject = getURLObject(url);
+	const domain = await resolveUrlAndHost(urlObject);
+	const matchRegion = domain.match(RE_AWS_ES_REGION);
+	const region = matchRegion[1] || defaultRegion;
+	const path = `${urlObject.pathname}${urlObject.search}`;
+	const protocol = urlObject.protocol;
+	return { domain, region, path, protocol };
 }
 
-function signedFetch(url, opts, creds) {
-	const { headers, path, protocol } = getSignableData(url, opts, creds);
-	opts.headers = headers;
+async function signedFetch(url, opts, creds) {
+	creds = creds || {};
+	// I need to get domain, region and path
+	const { domain, region, path, protocol } = await getSignableData(url);
+	opts.hostname = domain;
+	opts.headers.host = domain;
+	opts.headers['Content-Type'] = 'application/json';
+	opts.path = path;
+
+	// Create the HTTP request
+	const request = new HttpRequest(opts);
+
+	// Sign the request
+	const signer = new SignatureV4({
+		credentials: async () => creds,
```

Comment:
**suggestion:** I might be reading this wrong, but I don't think this needs to be a promise-based function because `creds` is always just plan data. The [Constructor for `SignatureV4`](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/classes/_aws_sdk_signature_v4.signaturev4-1.html#constructor) accepts either a `Credentials` object _or_ a `Provider<Credentials>`, and we already have a credentials object so I think this is equivalent:

```suggestion
		credentials: creds,
```

Reply:
I tested passing just the credentials and it didn't work. 
I tried first using plain credentials and later I changed for the async function. I don't know why but it's the way it is.

Reply:
🤷‍♂️  OK then, let's leave it